source/jobs.c

```diff
@@ -82,6 +82,24 @@ static const size_t apiTopicLength[] =
     JOBS_API_UPDATE_LENGTH + JOBS_API_FAILURE_LENGTH,
 };
 
+static const char * const jobStatusString[] =
+{
+    "QUEUED",
+    "IN_PROGRESS",
+    "FAILED",
+    "SUCCEEDED",
+    "REJECTED"
+};
+
+static const size_t jobStatusStringLengths[] =
+{
+    CONST_STRLEN( "QUEUED" ),
+    CONST_STRLEN( "IN_PROGRESS" ),
+    CONST_STRLEN( "FAILED" ),
+    CONST_STRLEN( "SUCCEEDED" ),
+    CONST_STRLEN( "REJECTED" )
```

Comment:
Having copied the strings here instead of somehow using the original struct `jobStatusString` seems like a potential place where a bug can creep in later on if any modification to the statuses are required.

Maybe we should assert that these sizes match with what we expect when the `init` function is called?

I leave it up to the author's judgement if there is good reason to have it like this.

Reply:
Yeah - This has bugged me and your comment has forced me to look deeper. Turns out `strlen` is an acceptable use here since the strings will always have a null terminator (per rule 21.17 in the 2012 edition). I'm going to switch to this since it's much more readable.